File: client/src/api/portfolio.ts
import { apiFetch } from '@/api/client'

export interface PositionResponse {
  symbol: string
  quantity: number
  avg_cost_basis: number
  total_cost: number
  current_price: number
  market_value: number
  unrealized_pnl: number
  unrealized_pnl_percent: number
  realized_pnl: number
}

export interface PortfolioSummary {
  total_value: number
  cash_balance: number
  invested_value: number
  total_return: number
  total_return_percent: number
}

export interface PortfolioResponse {
  summary: PortfolioSummary
  positions: PositionResponse[]
}

export function fetchPortfolio(): Promise<PortfolioResponse> {
  return apiFetch('/portfolio')
}

File: server/app/schemas/portfolio.py
from pydantic import BaseModel


class PositionResponse(BaseModel):
    symbol: str
    quantity: int
    avg_cost_basis: float
    total_cost: float
    current_price: float
    market_value: float
    unrealized_pnl: float
    unrealized_pnl_percent: float
    realized_pnl: float


class PortfolioSummary(BaseModel):
    total_value: float
    cash_balance: float
    invested_value: float
    total_return: float
    total_return_percent: float


class PortfolioResponse(BaseModel):
    summary: PortfolioSummary
    positions: list[PositionResponse]
